logstash-core/src/main/java/org/logstash/health/PipelineIndicator.java

```diff
@@ -96,11 +109,52 @@ public void serialize(final Details details,
                                   final SerializerProvider serializerProvider) throws IOException {
                 jsonGenerator.writeStartObject();
                 jsonGenerator.writeObjectField("status", details.getStatus());
+                FlowObservation flow = details.getFlow();
+                if (flow != null && !flow.isEmpty()) {
+                    jsonGenerator.writeObjectField("flow", flow);
+                }
                 jsonGenerator.writeEndObject();
             }
         }
     }
 
+    @JsonSerialize(using = FlowObservation.JsonSerializer.class)
+    public static class FlowObservation {
+        private static FlowObservation EMPTY = new FlowObservation(Map.of());
+        Map<String, Map<String,Double>> capture;
```

Comment:
L133 and L146 tries to access `capture`, making it empty would be safer.
```suggestion
        private Map<String, Map<String,Double>> capture = Map.of();
```

Reply:
It will never be `null` if it is final, since the constructor guards :)

```suggestion
        final Map<String, Map<String,Double>> capture;
```